### ISSUE TITLE
AltPalette's behavior Overlay and Remap issue

### ISSUE THREAD

**Issue by MirceaOfRivia** [Phobos-developers/Phobos#1112]
### Description

Animations using `AltPalette=yes` are always drawn over all layers resulting in the animations playing over the shroud and fogofwar.

The remap colors default to orange instead of red (as in the unittem.pal) or the house color of the player that triggered them. 
Information from RemapAnim is not transferred from animation to animation.

Example:
Unite (House Blue) - generate animation one with the remap (blue), expires with animation two (orange) house color is lost.
Changing AltPalette=yes to CustomPalette=unittem.pal makes the remap section the color of the pal (red) like the original and the animation respects the Layer= tag.
![anim](https://github.com/Phobos-developers/Phobos/assets/85160142/a01f4428-930b-4c08-bda0-ebea9e8cea06)

### Phobos Version

https://github.com/Phobos-developers/Phobos/releases/tag/build-36

### INI code

```ini
[E1]
DestroyAnim.Types=BODYDEB
DestroyAnim.Owner=Invoker
DestroyAnim.Random=yes

[BODYDEB]
Elasticity=0.0
MinZVel=7.0  ;Body
MinZVel=25.0  ;Body
ExpireAnim=GLI09DDED
LoopStart=3
LoopEnd=11
LoopCount=-1
RandomRate=125,150
Bouncer=yes
AltPalette=yes
Remapable=yes
Shadow=yes

[BODYIMPCT]
Report=TechnicalBodyfall
AltPalette=yes
Remapable=yes
LoopStart=20
LoopEnd=29
LoopCount=5
Rate=300
Layer=Surface
Shadow=yes
```

### Steps to reproduce

1. Unit dies and creates the animation with the Invoker tag
2. Animation 1 plays as Invoker tag Expiers to Animation 2
3. Animation 2 defaults to a new unknown tag
...

### Expected behaviour

The house information and the layer information should be transferable.
![deathFLIP](https://github.com/Phobos-developers/Phobos/assets/85160142/9b937fd5-1604-479d-bc1b-ed23bccb5468)

### Actual behaviour

The last animation created with the tag "ExpireAnim=" will ignore the layer tag and will be drawn on top of everything and the Remap color will default to an orange
![anim](https://github.com/Phobos-developers/Phobos/assets/85160142/f06a0372-5eca-423e-bf4e-9a57228b8d70)

### Checklist

- [X] The issue happens on the **latest official** version of Phobos and wasn't fixed yet.
- [X] I agree to elaborate the details if requested and provide thorough testing if the bugfix is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, gifs, videos, debug and exception logs, etc)*.

**Comment by Starkku** [comment]
Gonna clear some misconceptions here.

First of all, this is not a Phobos bug. No Phobos feature is involved and Phobos has not broken anything here.

Second, what `AltPalette` actually does is assign `LightConvertClass` (this and `ConvertClass` it is inherited from are responsible for determining which colors (palette) the object is drawn with as well as initializing and picking blitters  that handle the actual drawing on surface. `LightConvertClass` does in addition handle lighting adjustments and remapping to color scheme) that belongs to the first color scheme in internal color schemes array. 

Internally, the color schemes from `[Colors]` are duplicated, with first one being a single-shade version and the one following that is the fully shaded color scheme. So what `AltPalette` does is use the `LightConvertClass` from the single-shaded version of the first color scheme from `[Colors]`. Always, no exceptions. The drawing over shroud thing is probably some sort of blitter-related error that only occurs with the single-shade color scheme drawers.

Bottom line is that `AltPalette` does not and has never taken the animation's owner into accord in any way.

PR #1075 has `RemapAnim` that does what `MakeInfantry` or `CreateUnit.RemapAnim` do without any of the additional requirements, but it also only works for animations where this is specifically enabled, which is most of them but not all (exceptions that come to mind are infantry death anims, spawned missile trailers). It would be possible to change that to simply 'fixing' `AltPalette` which would make it work everywhere where the animation has an owner, but I am not sure if someone, somewhere relies on those anims drawing over shroud.

**Comment by mevitar** [comment]
As far as drawing over shroud is concerned, i think it should be its own thing and something that people can customize, not something that looks more like an obscure side effect. Fixing AltPalette and and allowing people to specify which animations can appear over the shroud is probably the best approach.

Alternatively, RemapAnim= could replicate what AltPalette= does, except with proper remapping and without the drawing-over-shroud part.

**Comment by Starkku** [comment]
This only happens if the game is launched without a spawner. Not sure why, will investigate.

**Comment by Starkku** [comment]
As of 963b54d this issue (`AltPalette` causing animations to draw over shroud) has been fixed. The remap color part is not a bug, but it has been changed regardless to better accomodate most intended use cases.

Excerpt from Phobos documentation for reference:
> Animations using AltPalette are now remapped to their owner’s color scheme instead of first listed color scheme and no longer draw over shroud. Color scheme from [AudioVisual] -> AnimRemapDefaultColorScheme is used if anim has no owner, which defaults to first listed color scheme from [Colors] still.
>  • They can also have map lighting apply on them if AltPalette.ApplyLighting is set to true.

Closing issue as completed.
